Extension/src/LanguageServer/client.ts

```diff
@@ -3972,6 +3998,17 @@ export class DefaultClient implements Client {
         compilerDefaults = await this.requestCompiler(path);
         DebugConfigurationProvider.ClearDetectedBuildTasks();
     }
+
+    public async showCopilotHover(content: string): Promise<ShowCopilotHoverResult> {
+        const params: ShowCopilotHoverParams = {content: content};
+        await this.ready;
+        return this.languageClient.sendRequest(ShowCopilotHoverRequest, params);
```

Comment:
It's common for multiple hover requests to be issued and get backlogged as the user moves the mouse around.  As each one is generated, the prior one is cancelled.  I think the cancellation token needs to be plumbed up here, all the way to these calls to sendRequest, in order for the `lsp_manager` on the native side to handle cancellation.  Then you'd also want to check the resulting object (like we do in most providers that call sendRequest) to ensure it's the object you expect.  If it's missing a required field, it's been replaced with a cancellation object.

Reply:
I'm hoping that since we're no longer going back and forth from the lsp server, this shouldn't be an issue. I've added exception catching around the one call this still does make to the native side. Let me know if there is any additional cancellation I should be tracking still.

Reply:
It looks like this `sendRequest` was moved to `getRequestInfo` and doesn't have a cancellation token passed to it.  I think you'll want to capture the cancellation token from the original(?) hover request (or, maybe the subsequent hover request caused by `showCopilotContent`?), and pass that to `sendRequest` to communicate that cancellation to the native process. Even if you're not explicitly checking for cancellation (`!request.is_active()`) in your LSP message handler, this would allow the LSP manager itself to discard requests in queue that have already been cancelled and to discard the response for an operation already cancelled.